securitas/tests/unit/controller/test_password.py

```diff
@@ -9,3 +11,117 @@ def test_password_reset(client):
     page = BeautifulSoup(result.data, 'html.parser')
     assert page.title
     assert page.title.string == 'Password Reset - The Fedora Project'
+
+
+def test_missing_credentials(client):
+    """Verify that missing details are caught"""
+    result = client.post(
+        '/password-reset',
+        data={
+            "username": "",
+            "current_password": "",
+            "password": "",
+            "password_confirm": "",
+        },
+        follow_redirects=True,
+    )
+    assert result.status_code == 200
+    page = BeautifulSoup(result.data, 'html.parser')
+    messages = page.select(".flash-messages .red")
+    assert len(messages) == 1
+    assert messages[0].get_text(strip=True) == 'Please fill in all fields to reset your password'
+
+
+def test_non_matching_passwords(client):
+    """Verify that passwords that dont match are caught"""
+    result = client.post(
+        '/password-reset',
+        data={
+            "username": "jbloggs",
+            "current_password": "sdsds",
+            "password": "password1",
+            "password_confirm": "password2",
+        },
+        follow_redirects=True,
+    )
+    assert result.status_code == 200
+    page = BeautifulSoup(result.data, 'html.parser')
+    messages = page.select(".flash-messages .red")
+    assert len(messages) == 1
+    assert messages[0].get_text(strip=True) == 'Password and confirmation did not match.'
+
+
+def test_password_policy(client):
```

Comment:
You should probably use the `dummy_user` fixture, so you don't depend on a specific user existing in FreeIPA before the tests are run.
Also, this tests needs the VCR decorator.